### DISCUSSION TITLE
How to install without docker

### DISCUSSION THREAD

**Discussion by SyedAbbas01** [ory/kratos#1067] [Q&A] [answered]
I tried to install to a VM but it would not accept dockers virtualisation

If I am not mistaken in a production environment docker will not be used.

Can you please confirm how this can be setup without docker?

**Comment by aeneasr** [comment]
Habe you seen https://www.ory.sh/kratos/docs/install ?

**Reply by SyedAbbas01** [reply]
Yes I have but at the moment I need to install without docker and setup a postgres db - how will the schema be generated for instance?

**Comment by vinckr** [comment, marked as answer]
You can use Docker in a production environment as well, you just have to set it up accordingly. 

Check the [reference config](https://www.ory.sh/kratos/docs/reference/configuration/) for your settings for database and schema ect. 

Everything else depends on what you want to achieve. 

See the "[Deployment Fundamentals](https://www.ory.sh/docs/ecosystem/deployment)" for more pointers regarding database. 

If you are looking for a managed deployment, we can help you there as well. Reach out to jared@ory.sh, he can give you some more information on our hosted service.

**Reply by SyedAbbas01** [reply]
Thank you for the reply @vinckr.  Correct me if am mistaken - I was under the impression that having db's in docker was not suitable (even if the data was persisted):

![image](https://user-images.githubusercontent.com/38494273/107937653-20b08800-6f7c-11eb-81de-22faf27c6573.png)

From hydra config https://www.ory.sh/hydra/docs/configure-deploy/

At the moment this is just part of proof of concept for a charity - so wanted to run out of containers to show viability. Currently working pro bono to get the up and running. The end goal is to run all authentication/authorization via hydra and kratos. Feeling slightly out of my depth as my background is dev not infrastructure.

You guys have done fantastic job appreciate your help! - I think this will really pick up traction in the coming years.

**Reply by vinckr** [reply]
Yes that is correct, it is better to not run your database in docker in production. If anything goes wrong with Docker your whole database could go poof. 
It is ok to do it for development or PoC. 

Are you positive you need to be an OAuth2 server for your use case? If you are not 100% sure, be sure to check out [this document](https://www.ory.sh/hydra/docs/concepts/before-oauth2). 
If you are sure, forgive me, but often people charge at a (non-trivial) Hydra deployment, that is not needed for their use case.

**Reply by vinckr** [reply]
Hey @SyedAbbas01 ,
Just checking in if your question was answered or if there was still something unclear.
If not, could you select the right answer please?

Thanks a ton! 🐝
